docs/installation.rst

```diff
@@ -9,16 +13,252 @@ Fantastic! Thank you!
 
 Here are some instruction to get your local coding environment set up.
 
-Dependencies
-============
+Setting Up
+==========
 
-BookBrainz is composed of:
+BookBrainz depends on having PostgreSQL, Redis, Elasticsearch and NodeJS set up and running.
 
+But first some basic configuration common to both docker and manual installation.
 
-* a NodeJS-based web server
-* a PostgresQL database
-* a Redis server to store user sessions
-* an ElasticSearch server to handle search
+Cloning
+*******
 
-The standard way to run these dependencies is using Docker.
-For a primer on what Docker is and how to use it, visit `DockerGettingStarted`_.
\ No newline at end of file
+To clone the repository and point the local HEAD to the latest commit in the stable branch, something like the following command should work
+
+::
+    
+    git clone --recurse-submodules https://github.com/metabrainz/bookbrainz-site.git
+
+Since this project makes use of git submodules, you need to use ``git clone --recurse-submodules`` to clone it.
+
+Alternatively, to manually initialize submodules, run these two commands
+::
+
+    git submodule init
+    git submodule update
+
+Currently we are using these submodules
+
+* `MonkeyDo/Lobes`_ in ``src/client/stylesheets/lobes``
+
+Configuration
+*************
+
+Create a copy of ``config/config.json.example`` and rename it to ``config.json``. You can do this by running this command from the bookbrainz-site directory.
+
+::
+
+    cp config/config.json.example config/config.json
+
+If you want to be able to sign-up and edit, you will need to set up authentication under ``musicbrainz``.
+To get the ``clientID`` and ``clientSecret`` tokens, head to `MusicBrainz`_ and register a new developer application.
+
+Make sure to enter the callback URL as ``http://localhost:<port>/cb`` (port: 9099 by default).
+
+You can then copy the tokens for that developer application and paste as strings in your ``config/config.json``. The tokens and callback URL in your ``config/config.json`` needs to match exactly the one for the developer application.
+
+Docker Installation
+===================
+The easiest way to get a local developement server running is using Docker. This will save you a fair amount of set up.
+
+You'll need to install Docker and Docker-compose on your development machine:
+
+* `Docker`_
+*  `Docker-compose`_
+
+When it is installed, follow the below instructions step by step.
+
```

Comment:
I would love to see another note here (maybe an `.. important::` one even !), about Windows users to *highly recommend* using WSL (Windows Subsystem for Linux) if it is available to them.
Are you familiar with it yourself?
This makes running Docker more efficient, and solved multiple compatibility issues some contributors have had in the past on Windows.

And here's a probably useful link: https://docs.microsoft.com/en-us/windows/wsl/about

And a definitely useful one on using Docker with WSL: https://docs.microsoft.com/en-us/windows/wsl/tutorials/wsl-containers

Reply:
I'm afraid, I have never tried running docker inside windows myself but still i'll add the WSL advice and if required we can add other instructions in the troubleshooting section.